### PR TITLE
docs: Update Java integration page

### PR DESCRIPTION
**Pull request by vikram-dagger**

### CODE REVIEW

File: docs/current_docs/integrations/java.mdx [hunk 1]
```diff
@@ -7,12 +7,36 @@ import TabItem from '@theme/TabItem';
 
 # Java
 
-Dagger can be used to perform common CI tasks - testing, containerizing, publishing and more - for any Java application.
+Dagger can be used to perform common CI tasks - testing, containerizing, publishing and more - for any Java application. This allows Java developers to create application delivery pipelines with the same language and tooling they use for their Java applications, and also benefit from faster local runs by using Dagger to cache Maven dependencies.
```
Comment (jpadams) [inline docs/current_docs/integrations/java.mdx:10]: Developers can use the Java SDK today to write a custom client (Cloak style), but Modules/Functions are not yet implemented. 

Additionally, the Java module we mention here is for use with module/function-able SDKs like TS, Go, Python and not Java today.

So there are two separate "integrations" today, not one. For this one:

```suggestion
Dagger can be used to perform common CI tasks - testing, containerizing, publishing and more - for any Java application in two ways. Java developers can create application delivery pipelines with the experimental Java SDK, using the same language and tooling they use for their Java applications, and benefitting from faster local runs by using Dagger to cache Maven dependencies. Note: The Java SDK is currently experimental and is moving soon to support Modules and Functions.

Additionally, if developers are using a Dagger SDK that supports Modules and Functions, such as the TypeScript, Python, or Go SDKs, they can leverage an excellent Java module from the [Daggerverse](https://daggerverse.dev/mod/github.com/jcsirot/daggerverse/java) which provides various Dagger Functions to work with Java projects. These Dagger Functions make it easy to configure the Java/Maven versions to use, the Maven sub-commands for building and packaging the application, and the output target.
```

File: docs/current_docs/integrations/java.mdx [hunk 2]
```diff
@@ -7,12 +7,36 @@ import TabItem from '@theme/TabItem';
...
 
-The following code sample demonstrates how these CI tasks can be encapsulated as Dagger Functions in a Dagger module. It assumes:
+## How it works
 
-- A Spring Boot Web application with Maven for dependency management
-- Credentials to publish the containerized application image to a registry
+The integration depends on the [`java` Dagger module](https://daggerverse.dev/mod/github.com/jcsirot/daggerverse/java), which provides various Dagger Functions to work with Java projects. These Dagger Functions make it easy to configure the Java/Maven versions to use, the Maven sub-commands for building and packaging the application, and the output target.
```
Comment (jpadams) [inline docs/current_docs/integrations/java.mdx:14]:
```suggestion
The integration depends on the [`java` Dagger module](https://daggerverse.dev/mod/github.com/jcsirot/daggerverse/java) and is currently supported from SDKs that support Dagger Modules and Functions.
```

**Review by jpadams — CHANGES_REQUESTED** [review]
I think we need to highlight the two ways folks need to interact with Dagger x Java today.

- Writing client SDK style in Java
- Using the `Java` Dagger module with an SDK that supports Modules/Functions (not Java yet)

**Comment by vikram-dagger** [comment]
> I think we need to highlight the two ways folks need to interact with Dagger x Java today.
> 
> * Writing client SDK style in Java
> * Using the `Java` Dagger module with an SDK that supports Modules/Functions (not Java yet)

I have modified your original suggestion taking into account @levlaz feedback in https://github.com/dagger/dagger/pull/8321. See https://github.com/dagger/dagger/pull/8321#issuecomment-2329766949. Lmk what you think!

**Review by jpadams — APPROVED** [review]
Ship it! 🚀

**Pull request merged**